pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -583,6 +533,19 @@ func podSetUpdates(wl *kueue.Workload, pr *autoscaling.ProvisioningRequest) []ku
 	})
 }
 
+func (c *Controller) deactivateWorkload(ctx context.Context, wl *kueue.Workload, msg string) error {
+	updatedWl := &kueue.Workload{}
+	if err := c.client.Get(ctx, client.ObjectKeyFromObject(wl), updatedWl); err != nil {
+		return err
+	}
+	workload.SetEvictedCondition(updatedWl, kueue.WorkloadEvictedByDeactivation, msg)
```

Comment:
Only set evicted condition if it is not set yet. Analogous as we do [here](https://github.com/kubernetes-sigs/kueue/blob/2b5184c3b1da933321b1c52795f4b2e04094c7ec/pkg/controller/core/workload_controller.go#L184-L214). Note that a workload may get evicted due to another reason in the meanwhile.

Reply:
Fixed